File: src/client/acontext-py/src/acontext/client_types.py
"""
Common typing helpers used by resource modules to avoid circular imports.
"""

from collections.abc import Mapping, MutableMapping
from typing import Any, BinaryIO, Protocol


class RequesterProtocol(Protocol):
    def request(
        self,
        method: str,
        path: str,
        *,
        params: Mapping[str, Any] | None = None,
        json_data: Mapping[str, Any] | MutableMapping[str, Any] | None = None,
        data: Mapping[str, Any] | MutableMapping[str, Any] | None = None,
        files: Mapping[str, tuple[str, BinaryIO, str | None]] | None = None,
        unwrap: bool = True,
    ) -> Any:
        ...

File: src/client/acontext-ts/src/client-types.ts
/**
 * Common typing helpers used by resource modules to avoid circular imports.
 */

export interface RequesterProtocol {
  request<T = unknown>(
    method: string,
    path: string,
    options?: {
      params?: Record<string, string | number>;
      jsonData?: unknown;
      data?: Record<string, string>;
      files?: Record<string, { filename: string; content: Buffer | NodeJS.ReadableStream; contentType: string }>;
      unwrap?: boolean;
    }
  ): Promise<T>;
}

